### DISCUSSION TITLE
Incorrect use of <label for=FORM_ELEMENT> on SelectOneMenu

### DISCUSSION THREAD

**Discussion by pierrefroy** [primefaces/community#1062] [PrimeFaces] [answered]
Hi,

I'm trying to find the ause of a bug I'm getting on dialogs when the close on escape is not working. I can get the escape key to close dialogs by using this property: closeOnEscape="true". Except on some dialogs where it doesn't work for an obscure reason. 

I used the browser console to see if there is a difference between working and non broken dialogs. What I'm seeing for now is that dialogs with broken escape are displaying errors for label when the label is applied to SelectOneMenu. Here is an example of such case.

The error message is : `Incorrect use of <label for=FORM_ELEMENT>
`
```
<!-- SELECT FACILITY -->
<div class="field col-12 md:col-12">
    <p:outputLabel value="#{msg.Facility_label}"/>
    <p:message for="facility" display="text"/>
    <p:selectOneMenu id="facility" value="#{facilityController.selected}"
                     converter="omnifaces.SelectItemsConverter"
                     height="35vh">
        <p:ajax event="itemSelect"
                listener="#{faultController.onSelectFacility}"
                update="facilityLocation"/>
        <f:selectItems value="#{facilityController.itemsLookup}" var="facility"
                       itemLabel="#{facility.name}" itemValue="#{facility}"/>
        <f:facet name="footer">
            <h:outputFormat value="#{msg.Facility_count}"
                            style="font-weight:bold;" styleClass="block px-2 py-1 surface-200">
                <f:param value="#{facilityController.itemsLookup.size()}"/>
            </h:outputFormat>
        </f:facet>
    </p:selectOneMenu>
</div>
```
Thanks for your help.

Primefaces 13.0.5

Pierre

**Comment by melloware** [comment]
@pierrefroy are you saying CloseOnEscape for the dialog doesn't work in the above scenario?  If so please 
create an executable example using the [PrimeFaces Test](https://github.com/primefaces/primefaces-test) project.  and if you can reproduce open a PrimeFaces GitHub Issue please!

**Reply by pierrefroy** [reply]
Hi @melloware ,

Yes, this cause the close on escape key to stop working when the form in my dialog contains labels for SelectOneMenu. If I remove all the fields in the form, the escape key start working again.

Interestingly I had the idea this morning to revert the Primefaces version from 13.0.5 to 13.0.à4 and it solved my issue. So I can confirm that this bug happens only with Primefaces 13.0.5.

I'll try to reproduce the issue with the Primefaces Test, but it might be hard as a lot of the code in my form is domain related to my current project.

Best regards,
Pierre Francis

**Reply by melloware** [reply]
Hmm here are what was in 13.0.5: https://github.com/primefaces/primefaces/releases/tag/13.0.5

**Reply by melloware** [reply]
Thinking the issue is this: https://github.com/primefaces/primefaces/issues/11308

Or this: https://github.com/primefaces/primefaces/issues/11282

Only two related issues

**Reply by melloware** [reply, marked as answer]
Reported and fixed for 13.0.6: https://github.com/primefaces/primefaces/issues/11369

**Reply by pierrefroy** [reply]
Hello @melloware ,

I had a look at 11308 and it looks like it. As of 11282, I don't know, I would have to investigate in more details. I'll wait for the fix on 13.0.6 and keep you informed of the results. Thank you for your quick feedback and your responsiveness.

Regards,
Pierre Francis

**Reply by melloware** [reply]
I was able to reproduce it @pierrefroy basically if the SelectOneMenu had focus it was "eating" the ESC key.  I have it so it only respect the ESC key if the SelectOneMenu is open to close the SelectOneMenu.  Not when its closed it will pass the ESC key up the chain and close.

**Reply by pierrefroy** [reply]
I'm glad you could find it swiftly. This explains the problems with some of my dialogs with plenty of SelectOneMenu, while the ones with only InputText are working perfectly.

Thanks again and have a nice day ;)
